proposals/lip-0055.md

```diff
@@ -68,6 +69,15 @@ The schema for the block assets allows each module to include its serialized dat
 
 Each entry of the block assets is then inserted in a Merkle tree, whose root is included in the block header as the `assetsRoot` property. Inserting the assets root rather than the full assets allows to bound the size of the block header while still authenticating the content of the block assets.
 
+
+### Failing and Invalid Transactions
+
+In the current protocol, a transaction that is included in a block always has its intended consequences. In particular, if an error occurs during the its processing, the transaction is simply invalid and it is not included in the block payload.
+
+With this LIP, we introduce failing transactions. A failing transaction is one for which an error occurs but which is nevertheless included in a block. 
+In particular, if the error happens after the transaction has been verified and the state transitions happening before the *command execution* have been processed (see the ["Block processing stages"](#block-processing-stages) section below), the transaction is considered to be *failed*. Otherwise, the transaction is *invalid* and it is not included in the block (in a way analogous to the current protocol).
```

Comment:
should this section also say `afterCommandExecute` will not make transaction fail but make it invalid, or it's implied?

Reply:
Sorry, I forgot to push, check it now